lib/spack/spack/util/remote_file_cache.py

```diff
@@ -76,17 +77,23 @@ def local_path(raw_path: str, sha256: str, make_dest: Optional[Callable[[], str]
     if not raw_path:
         raise ValueError("path argument is required to cache remote files")
 
+    file_schemes = ["", "file"]
+
     # Allow paths (and URLs) to contain spack config/environment variables,
     # etc.
+    win_path = pathlib.PureWindowsPath(raw_path)
+    if win_path.drive:
```

Comment:
Tests are current failing because of the order of operations here.
I like this trick, it's clever, but it's being run on a completely different string than what's being passed to urllib. The additions to `canonicalize_path` will add a drive from CWD if one does not currently exist in the string. But because we're checking for a drive before we artificially add one in the case of a relative path, the `if url.scheme in file_schemes` check is failing to catch that we have a drive component as a scheme, and failing.

Reply:
As I mentioned off-line I'll revisit this. There was a reason I had for doing that which I don't recall off-hand.

Reply:
(I think got moved around, but not in a way that marked the comment as outdated)